dashboard/lib/pd/jot_form/form_questions.rb

```diff
@@ -63,9 +63,11 @@ def [](id_or_name)
       # Note: matrix questions are expanded into their sub-questions,
       #   so the resulting summary may contain more items than the original list.
       # See #Question::summarize
-      def summarize
+      # @param show_hidden_questions [Boolean] optional, default false
+      def summarize(show_hidden_questions: false)
         {}.tap do |summary|
           @questions_by_id.values.sort_by(&:order).each do |question|
+            next if question.hidden? && !show_hidden_questions
```

Comment:
nit - suggest switching the order of this conditional. I think it's slightly faster with lazy evaluation

Reply:
Actually, I'd like to keep this. Reordering it will be harder to read, and `hidden?` is just [aliasing](https://github.com/code-dot-org/code-dot-org/blob/d37236b428091be4aff59ca05b165960ef209755/dashboard/lib/pd/jot_form/question.rb#L18) an [accessor](https://github.com/code-dot-org/code-dot-org/blob/d37236b428091be4aff59ca05b165960ef209755/dashboard/lib/pd/jot_form/question.rb#L9), so there's very little overhead - just returning the value of the `@hidden` member variable.

Reordering would look like one of these:
```ruby
# against style rules, if !
next if !show_hidden_questions && question.hidden? 

# harder to reason about
next unless show_hidden_questions || !question.hidden?
```

I suppose we could rearrange it in the positive, like this, but I also find this less clear:
```ruby
if show_hidden_questions || !question.hidden
  summary.merge! question.summarize
end
```